src/app/pages/courses-page/courses-page.component.ts

```diff
@@ -44,6 +52,12 @@ export class CoursesPageComponent implements OnInit {
   }
 
   private getCourses() {
-    this.courses$ = this.coursesService.getCourses$();
+    this.coursesService.getCourses$().subscribe(
```

Comment:
let's use streams and `async` pipe without `subscribe`...
I'd consider updating workflow of this component by passing search criteria to `coursesService` and getting necessary stream of courses

Reply:
it is not very optimized solution to store two similar arrays...

Reply:
Agree, will do it